### DISCUSSION TITLE
Feature request - Adding PAT support for the Umbraco Cloud git

### DISCUSSION THREAD

**Discussion by timknightkrow** [umbraco/Umbraco.Cloud.Issues#797] [Ideas]
### Issue description

As you cannot guarantee the persistance of non-master git branches, we would like to use an external repo to manage the source ande branches.

We are aware of an approach using Dev Ops that allows changes commited to an azure repo to be pushed to cloud, but this solution only works one way, with any changes made on the cloned dev environment not automatically pulling down again to that external repo.

The solution for us would be an Azure Dev Ops repo with bidirectional mirroring (specifically this as opposed to GitLab).

In order to do this, a plugin is available https://marketplace.visualstudio.com/items?itemName=swellaby.mirror-git-repository

However, this requires a PAT (personal access token) in order to link up with the Umbraco Cloud git repository.

After speaking with support, they have suggested I raise a feature request to ask for the ability to obtain/generate a PAT from the Umbraco Cloud portal.

Could this be implemented? I'm sure there are other good uses for a PAT beyond this, where it is not suitable to use a password directly. PAT also allow access to be revoked without having to change the account password every time.

Many thanks

**Comment by lakesol** [comment]
This would be very useful.

**Comment by sajumb** [comment]
Sorry for the late response to this feature request. We had to wait until after the first day of our Codegarden conference for a response to not tease the upcoming CI/CD support that soon will be out in beta. But something very relevant is coming... stay tuned. Or see the sessions in the Codegarden app. We are talking about the CI/CD feature in both the keynote and the "Umbraco Cloud & You" sessions.

**Comment by eqtr-ab** [comment]
Is there any update on this @sajumb? OP's point about being able to mirror the cloud repo to somewhere like Azure devops without having to pay for Gitlab would be beneficial.

**Comment by mclausen** [comment]
Hi @eqtr-ab, and thank you for reaching out :)

A PAT token would definitely be useful and is something we want to implement. We are still refining a modernized version of handling users and permissions for the future, but it is still very much a whiteboard phase of those discussions. Public Access Tokens have definitely been part of the internal discussion; however, there are no promises on when we will get around to that.

In the meantime, we have a fully supported CI/CD setup that you can read about [here](https://docs.umbraco.com/umbraco-cloud/set-up/project-settings/umbraco-cicd). This should help you get started integrating your favorite CI/CD tool with Umbraco Cloud.

I will move this issue into discussions, where we can exchange ideas for future support of Public Access Tokens :)
